### PR TITLE
Feature/add user

### PR DESCRIPTION
**Pull request by Gimleux**

### CODE REVIEW

File: webapp_frontend/src/assets/images/icons/material.io/check_circle-24px.svg [hunk 1]
```diff
@@ -0,0 +1 @@
+<svg xmlns="http://www.w3.org/2000/svg" height="24" viewBox="0 0 24 24" width="24"><path d="M0 0h24v24H0z" fill="none"/><path d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z" fill="green"/></svg>
```
Comment (qvalentin) [inline webapp_frontend/src/assets/images/icons/material.io/check_circle-24px.svg:1]: alternativly we could use: https://icons.getbootstrap.com/

File: webapp_frontend/src/components/pages/Registration.tsx [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+                                <span className={passwordInformationLength ? "text-success" : "text-muted"}>Passwords must be at least 8 characters.</span>
+                            </div>
+                            <div>
+                                <img alt={"status icon password contains uppercase character"} src={passwordInformationUppercase ? check_svg : info_svg}/>
+                                <span className={"sr-only"}>{passwordInformationUppercase ? "Done: " : "Missing: "}</span>
+                                <span className={passwordInformationUppercase ? "text-success" : "text-muted"}>Passwords must be at least contain 1 uppercase character.</span>
+                            </div>
+                            <div>
+                                <img alt={"status icon password contains lowercase character"} src={passwordInformationLowercase ? check_svg : info_svg}/>
+                                <span className={"sr-only"}>{passwordInformationLowercase ? "Done: " : "Missing: "}</span>
```
Comment (qvalentin) [inline webapp_frontend/src/components/pages/Registration.tsx]: much inclusive much wow

**Review by qvalentin — APPROVED** [review]
nice

**Review by qvalentin — CHANGES_REQUESTED** [review]
Selecting the group(s) should also be possible.

**Comment by Gimleux** [comment]
> 
> 
> Selecting the group(s) should also be possible.

Will be discussed in next meeting

File: webapp_frontend/src/background/api/registration.ts [hunk 3]
```diff
@@ -1,7 +1,41 @@
...
+                }
+                resolve(response);
+            })
+            .catch((error: AxiosError) => {
+                const response: IRegisterServerResponse = {
+                    httpStatus: error.response!.status,
+                    httpMessage: error.response!.statusText
+                }
+                if (error.response!.status === 409) {
+                    response.outputMessage = "Username is already taken."
```
Comment (qvalentin) [inline webapp_frontend/src/background/api/registration.ts]: why not take the text the BE returns: error.response?.data.message

File: webapp_frontend/src/components/pages/Registration.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+                setAlertVisibility(false)
+            }, duration)
+        }
+    }
+
+    const handlePasswordChange = (event: ChangeEvent<HTMLInputElement>) => {
+        event.preventDefault();
+        let value = event.target.value;
+        value = deleteSpaces(value);
+        setPasswordInformationLength(!notMinStrLength(value, MIN_PASSWORD_LENGTH));
```
Comment (qvalentin) [inline webapp_frontend/src/components/pages/Registration.tsx:88]: max lenght for the BE is 20, so this should also be here. @open-schnick but what if i want my password to be longer than 20 chars?

**Review by qvalentin — CHANGES_REQUESTED** [review]

File: webapp_frontend/src/components/pages/Registration.tsx (hunk 4 above)
Comment (Gimleux) [reply webapp_frontend/src/components/pages/Registration.tsx:88]: done

Comment (open-schnick) [reply webapp_frontend/src/components/pages/Registration.tsx:88]: @qvalentin well we could implement a config key. How bout roadmap?

File: webapp_frontend/src/components/pages/Registration.tsx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+                            <Form.Label>Password</Form.Label>
+                            <Form.Control type="password"
+                                          placeholder="Must contain one number, uppercase & lowercase letter each"
+                                          value={password}
+                                          onChange={(event: ChangeEvent<HTMLInputElement>) => handlePasswordChange(event)}/>
+                            <div>
+                                <img alt={"status icon password length"}
+                                     src={passwordInformationLength ? check_svg : info_svg}/>
+                                <span className={"sr-only"}>{passwordInformationLength ? "Done: " : "Missing: "}</span>
+                                <span className={passwordInformationLength ? "text-success" : "text-muted"}>Passwords must be at least 8 characters.</span>
```
Comment (qvalentin) [inline webapp_frontend/src/components/pages/Registration.tsx]: and max 20

**Review by qvalentin — APPROVED** [review]
just change the text, then we can merge

**Comment by qvalentin** [comment]
the password check kind of sucks, I cant past my password, and I also can't remove the first letter that I put in

**Comment by Gimleux** [comment]
> 
> 
> the password check kind of sucks, I cant past my password, and I also can't remove the first letter that I put in

you just cannot paste a password if it is > maxLength
There should be an error visible

**Comment by Gimleux** [comment]
> I also can't remove the first letter that I put in

solved

**Review by open-schnick — APPROVED** [review]

**Pull request merged**